File: 2019/days/d2.cs
namespace AOC {
    class Day2 : AoCDay {
        public Day2(string linesRaw): base(linesRaw) {}
        
        private IntComputer comp = new IntComputer();
        override public string Part1() {
            comp.Load(lines[0]);
            comp.Run(12, 2);
            return comp.ins[0].ToString();
        }

        override public string Part2() {
            foreach(int n in Enumerable.Range(0, 100)) {
                foreach(int v in Enumerable.Range(0, 100)) {
                    comp.Load(lines[0]);
                    comp.Run(n, v);
                    if(comp.ins[0] == 19690720) {
                        return (100 * n + v).ToString();
                    }
                }
            }
            return "Couldn't find";
        }
    }
}

File: 2021/Ruby/days/d2.rb
module AOC 
    class Day2 < AocDay
        def reset
            @depth = 0
            @horiz = 0
            @aim = 0
        end

        def part1()
            reset()
            @lines.each{|l| 
                ins, amt = l.split
                amt = amt.to_i

                case ins
                in 'forward'
                    @horiz+=amt
                in 'down'
                    @depth+=amt
                in 'up'
                    @depth-=amt
                end
            }
            @p1 = @depth*@horiz
        end

        def part2()
            reset()
            @lines.each {|l|
                ins, amt = l.split
                amt = amt.to_i

                case ins
                in 'forward'
                    @horiz+=amt
                    @depth+=@aim*amt
                in 'down'
                    @aim += amt
                in 'up'
                    @aim -= amt
                end
            }
            @p2 = @depth*@horiz
        end
    end
end